provider/azure/internal/imageutils/images.go

```diff
@@ -81,22 +80,91 @@ func SeriesImage(
 	}, nil
 }
 
-func offerForUbuntuSeries(series string) (string, string, error) {
-	seriesVersion, err := jujubase.SeriesVersion(series)
-	if err != nil {
-		return "", "", errors.Trace(err)
+// supportedOldStyleUbuntuBases is a slice of all bases juju
+// supports, which use the old-style offer id like
+// "0001-com-ubuntu-server-${series}".
+//
+// Recently Canonical changed the format for Azure images offer ids
+// and SKUs in Azure. The threshold for this change was noble, so if
+// we want to deploy bases before noble, we must branch and and use the
+// old format.
+//
+// The old format offer ids have format `0001-com-ubuntu-server-${series}`
+// or `001-com-ubuntu-server-${series}-daily` and have SKUs formatted
+// `${version_number}-lts`, `${version_number}-gen2`, etc.
+//
+// The new format offer ids have format `ubuntu-${version_number}`,
+// `ubuntu-${version_number}-lts`, `ubuntu-${version_number}-lts-daily`,
+// etc. and have SKUs `server`, `server-gen1`m, `server-arm64`, etc.
+//
+// Since there are only a finte number of Ubuntu versions we support
+// before Noble, we hardcode this list. So when new versions of Ubuntu
+// are
+//
+// All Ubuntu images we support outside of this list have offer
+// id like "ubuntu-${version}-lts" or "ubuntu-${version}"
+var oldStyleUbuntuBases = []jujubase.Base{
+	jujubase.MustParseBaseFromString("ubuntu@20.04"),
+	jujubase.MustParseBaseFromString("ubuntu@20.10"),
+	jujubase.MustParseBaseFromString("ubuntu@21.04"),
+	jujubase.MustParseBaseFromString("ubuntu@21.10"),
+	jujubase.MustParseBaseFromString("ubuntu@22.04"),
+	jujubase.MustParseBaseFromString("ubuntu@22.10"),
+	jujubase.MustParseBaseFromString("ubuntu@23.04"),
+	jujubase.MustParseBaseFromString("ubuntu@23.10"),
+}
+
+func ubuntuBaseIsOldStyle(base jujubase.Base) bool {
+	for _, oldBase := range oldStyleUbuntuBases {
+		if base.IsCompatible(oldBase) {
+			return true
+		}
 	}
-	seriesVersion = strings.ReplaceAll(seriesVersion, ".", "_")
-	return fmt.Sprintf("0001-com-ubuntu-server-%s", series), seriesVersion, nil
+	return false
 }
 
 // ubuntuSKU returns the best SKU for the Canonical:UbuntuServer offering,
 // matching the given series.
-func ubuntuSKU(ctx context.ProviderCallContext, series, stream, location string, client *armcompute.VirtualMachineImagesClient) (string, string, error) {
-	offer, seriesVersion, err := offerForUbuntuSeries(series)
+//
+// TODO(jack-w-shaw): Selecting the 'daily' stream is currently broken for old style
```

Comment:
Why is it broken? Is it out of our control? If not fix it in this PR or else it won't get done. If not, is there an upstream bug we can point to in the TODO.

Reply:
The reason I haven't fixed it here is because this PR is getting quite large, and I don't want to add too much

If I were to create a bug and assign it to myself, would that be ok?

The reason it's broken is because CPC upload their daily images to another offer + don't take SKUs with `daily` or `daily-lts` or anything, but a proper fix would requires a bit more investigation.

Reply:
Sure, that's fine. Please file the bug and reference it in the TODO.